### PR TITLE
Remove hard dependency on PDO (4.4)

### PR DESCRIPTION
**Pull request by fritzmg**
| Q                | A
| -----------------| ---
| Fixed issues     | Fixes #2459
| Docs PR or issue | -

This would add the PDO options dynamically via the Manager Plugin of the Manager Bundle. It will not add anything if a driver other than `pdo_mysql` is used, or custom `options` have already been set.

### CODE REVIEW

File: manager-bundle/src/ContaoManager/Plugin.php [hunk 1]
```diff
@@ -150,6 +151,28 @@ public function getExtensionConfig($extensionName, array $extensionConfigs, Plug
...
+
+        return $this->setPdoDriverOptions($extensionConfigs, $container);
+    }
+
+    /**
+     * Sets path to enable autoloading of legacy Contao modules.
+     *
+     * @param string $modulePath
+     */
+    public static function autoloadModules($modulePath)
```
Comment (leofeyer) [inline manager-bundle/src/ContaoManager/Plugin.php:163]: Why did you add this method?

Comment (bytehead) [reply manager-bundle/src/ContaoManager/Plugin.php:163]: It was already there, see lines 190-196 on the diff.

Comment (fritzmg) [reply manager-bundle/src/ContaoManager/Plugin.php:163]: It's already there:

https://github.com/contao/contao/blob/8b07746dea9eecb9527a34fe05cb8ff824cd7aa2/manager-bundle/src/ContaoManager/Plugin.php#L189-L197

Comment (fritzmg) [reply manager-bundle/src/ContaoManager/Plugin.php:163]: I originally inserted the new private methods before `public static function autoloadModules` - however, as per the code style rules, `public` methods come before `private` ones, therefore the CS fixer moved it around, resulting in this slightly more unreadable diff ;)

**Review by aschempp — DISMISSED** [review]

**Comment by fritzmg** [comment]
I'll provide a separate PR for Contao 4.9, as this cannot be merged upstream.

File: manager-bundle/src/ContaoManager/Plugin.php [hunk 2]
```diff
@@ -187,12 +210,41 @@ public function getExtensionConfig($extensionName, array $extensionConfigs, Plug
...
+        // Do not add PDO options, if constant does not exist
+        if (!\defined('PDO::MYSQL_ATTR_MULTI_STATEMENTS')) {
+            return $extensionConfigs;
+        }
+
+        foreach ($extensionConfigs as $extensionConfig) {
+            // Do not add PDO options, if selected driver is not pdo_mysql
+            if (isset($extensionConfig['dbal']['connections']['default']['driver']) && 'pdo_mysql' !== $extensionConfig['dbal']['connections']['default']['driver']) {
+                return $extensionConfigs;
+            }
```
Comment (fritzmg) [inline manager-bundle/src/ContaoManager/Plugin.php]: A subsequent extension config could set the driver back to `pdo_mysql`. Should we consider this possibility? Seems unlikely though.

Comment (aschempp) [reply manager-bundle/src/ContaoManager/Plugin.php]: Our only real problem is that the constants don't exist, but we check for that. So I'd be fine with that.

**Review by aschempp — DISMISSED** [review]

**Review by leofeyer — APPROVED** [review]

**Comment by leofeyer** [comment]
Thank you @fritzmg.
